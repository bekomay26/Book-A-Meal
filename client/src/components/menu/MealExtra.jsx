import React from 'react';
import PropTypes from 'prop-types';
import { Checkbox, Grid } from 'semantic-ui-react';

const MealExtra = ({ extra, onQtyChange, extraStatus, handleChecked, indexKey }) => {
  const extStatPosition = extraStatus.findIndex(extraStat => extraStat.key === extra.id);
  let extraQty = 1;
  if (extraStatus[extStatPosition] !== undefined) {
    extraQty = extraStatus[extStatPosition].qty;
  } else {
    extraQty = 1;
  }

  return (<div>
    <Grid divided="vertically">
      <Grid.Row>
        <Grid.Column width={8}>
          <Checkbox
            onChange={(event) => handleChecked(event, extra.id, extra, extraQty)}
            className="menu-modalextra-text"
            label={extra.title}
          />
        </Grid.Column>
        <Grid.Column width={4} className="menu-modalextra-qty-grid">
          <input
            type="number"
            className="col-3 menu-modalextra-qty"
            onInput={(event) => onQtyChange(event, extra.id, extra.price)}
            defaultValue={1}
            min={1}
          />
        </Grid.Column>
        <Grid.Column width={4}>
          <div className="menu-modalextra-text">
            &#x20A6;{extra.price * extraQty}
          </div>
        </Grid.Column>
      </Grid.Row>
    </Grid>
  </div>)
};

MealExtra.propTypes = {
  extra: PropTypes.shape({
    title: PropTypes.string.isRequired,
    price: PropTypes.number.isRequired,
  })
    .isRequired,
};

export default MealExtra;
