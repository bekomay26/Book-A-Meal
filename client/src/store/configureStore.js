import { createStore, applyMiddleware } from 'redux';
import reduxImmutableStateInvariant from 'redux-immutable-state-invariant';
import thunk from 'redux-thunk';
import rootReducer from '../reducers/index';


const configureStore = initialState =>
  createStore(
    rootReducer,
    initialState,
    applyMiddleware(thunk, reduxImmutableStateInvariant()),
  );

export default configureStore;
