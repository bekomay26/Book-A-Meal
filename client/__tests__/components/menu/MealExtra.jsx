import React from 'react';
import { shallow, mount, render } from 'enzyme';
import toJson from 'enzyme-to-json';
import configureStore from 'redux-mock-store';
import { Provider } from 'react-redux';
import thunk from 'redux-thunk';
import MealExtra from '../../../src/components/menu/MealExtra';
import { getExtrasResponse } from '../../__mocks__/mockExtras';

const middlewares = [thunk];
const mockStore = configureStore(middlewares);
const store = mockStore({
  menuReducer: {
    menu: [],
  },
  authReducer: {
    isAuthenticated: true,
    userName: '',
  },
});

const extraArray = getExtrasResponse.extras;
function setup() {
  const props = {
    extra: extraArray[0],
    onQtyChange: jest.fn(),
    extraStatus: [{ isChecked: false, extra: extraArray[0], key: 1, selectedExt: [], qty: 2 }],
    handleChecked: jest.fn(),
    indexKey: 1,
  };
  return shallow(<MealExtra {...props} />, reactrouter.get());
}

describe('The Menu Extra Component', () => {
  it('should render meal Extra component correctly', () => {
    const wrapper = setup();
    expect(toJson(wrapper)).toMatchSnapshot();
  });
  it('should render meal Extra component correctly', () => {
    const props = {
      extra: extraArray[0],
      onQtyChange: jest.fn(),
      extraStatus: [],
      handleChecked: jest.fn(),
      indexKey: 1,
    };
    const wrapper = mount(<Provider store={store}><MealExtra {...props} /></Provider>, reactrouter.get());
    const input = wrapper.find('input').get(1);
    expect(input.props.defaultValue).toEqual(1);
    wrapper.unmount();
  });

  it('should check when clicked', () => {
    const props = {
      extra: extraArray[0],
      onQtyChange: jest.fn(),
      extraStatus: [],
      handleChecked: jest.fn(),
      indexKey: 1,
    };
    const wrapper = mount(<Provider store={store}><MealExtra {...props} /></Provider>, reactrouter.get());
    const checkButton = wrapper.find('.menu-modalextra-text').first();
    expect(checkButton.length).toEqual(1);
    checkButton.simulate('click');
    expect(wrapper.find('.menu-modalextra-text').last().props().children[1]).toEqual(50);
    wrapper.unmount();
  });

  // Not working
  it('should change value on input', () => {
    const props = {
      extra: extraArray[0],
      onQtyChange: jest.fn(),
      extraStatus: [],
      handleChecked: jest.fn(),
      indexKey: 1,
    };
    const wrapper = mount(<Provider store={store}><MealExtra {...props} /></Provider>, reactrouter.get());
    const checkButton = wrapper.find('.menu-modalextra-qty-grid').first();
    expect(checkButton.length).toEqual(1);
    wrapper.find('input').last().simulate('click').simulate('keypress', { which: 2 });
    wrapper.find('input').last().simulate('change', { target: { value: 2 } });
    wrapper.unmount();
    // expect(wrapper.find('input').last().prop('value').target).toEqual(2);
    // expect(wrapper.find('input').get(1)).toEqual(2);
    // expect(wrapper.find('input').last().props().value).toEqual(2);
  });
});
