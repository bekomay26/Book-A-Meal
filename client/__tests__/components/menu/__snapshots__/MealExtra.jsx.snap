// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The Menu Extra Component should render meal Extra component correctly 1`] = `
<div>
  <Grid
    divided="vertically"
  >
    <GridRow>
      <GridColumn
        width={8}
      >
        <Checkbox
          className="menu-modalextra-text"
          label="beef"
          onChange={[Function]}
          type="checkbox"
        />
      </GridColumn>
      <GridColumn
        className="menu-modalextra-qty-grid"
        width={4}
      >
        <input
          className="col-3 menu-modalextra-qty"
          defaultValue={1}
          min={1}
          onInput={[Function]}
          type="number"
        />
      </GridColumn>
      <GridColumn
        width={4}
      >
        <div
          className="menu-modalextra-text"
        >
          â‚¦
          50
        </div>
      </GridColumn>
    </GridRow>
  </Grid>
</div>
`;
